Bug 2115899: BuildConfig form: Change image repository to image registry

jerolimov opened the pull request:
**Fixes**: 
https://bugzilla.redhat.com/show_bug.cgi?id=2115899

**Analysis / Root cause**: 
OpenShift is all about standard container images and standard container image registries, btw. called registries, not repositories.

See also https://docs.openshift.com/container-platform/4.10/registry/index.html

**Solution Description**: 
Updated the labels:

1. On "Import container image" page update the helptext below the `Image name from external registry` from `private repository` to `private registry`
    
3. On the BuildConfig ediit page:
    1. Updated `Build from` option `Docker image` to `External container image`.
        
        The other options for `Build from` are `None`, `Image Stream Tag`, and `Image Stream Image` are not changed.
    4. Updated `Push to` option `Docker image` to `External container image`.
        
        The other options for `Build from` are `None` and `Image Stream Tag` are not changed.
    3. When these options are selected there is shown the same input field below select.
        Changed this title from `Docker image repository` to `Image registry` for both!

**Screen shots / Gifs for design review**: 
| Page | Before | After |
|---|---|---|
| Add > Import from container | ![before-image](https://user-images.githubusercontent.com/139310/188722117-220253c5-1045-49a6-a043-daa5ba550c24.png) | ![after-image](https://user-images.githubusercontent.com/139310/188722121-ee14407f-326b-42cb-bfb3-881a261e5ea9.png) |
| Edit BuildConfig | ![buildconfig-before](https://user-images.githubusercontent.com/139310/188722111-fa3346e5-0fa3-4577-85a2-b6cfecee0856.png) | ![buildconfig-after](https://user-images.githubusercontent.com/139310/188722107-ca614c54-b304-4255-86ae-7e82087a0e2b.png) |

**Unit test coverage report**: 
Unchanged

**Test setup:**
Unchanged

**Browser conformance**: 
- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

/cc @serenamarie125

jerolimov commented:
/retest

jerolimov commented:
/retest

jerolimov commented on frontend/packages/dev-console/locales/en/devconsole.json:
```diff
@@ -112,11 +112,11 @@
   "Image stream tag": "Image stream tag",
   "Image stream image": "Image stream image",
-  "Docker image repository": "Docker image repository",
+  "Image repository": "Image repository",
```
Missed this, will update to Image registry here as well:

```suggestion
  "Image repository": "Image repository",
```

serenamarie125 commented:
LGTM!  As discussed, we should be using image registry, not image repository.  Could you update the screenshot you have for the Edit BuildConfig?  Thanks!

jerolimov commented:
> LGTM! As discussed, we should be using image registry, not image repository. Could you update the screenshot you have for the Edit BuildConfig? Thanks!

Updated the "image repository" to "image registry" on the BuildConfig edit page as well.

Updated the Solution Description section and screenshots in my first comment.

/cc @debsmita1 @invincibleJai

invincibleJai commented:
@jerolimov looks like frontend job needs some ❤️

invincibleJai commented on frontend/packages/dev-console/src/components/buildconfig/sections/ImagesSection.tsx:
```diff
@@ -89,18 +89,18 @@ const ImagesSection: React.FC<{}> = () => {
           imageStreamTag: t('devconsole~Image Stream Tag'),
           imageStreamImage: t('devconsole~Image Stream Image'),
-          dockerImage: t('devconsole~Docker image'),
+          dockerImage: t('devconsole~External container image'),
```
should image be Image[upper case I] as we have above `Image Stream Image`

![image](https://user-images.githubusercontent.com/5129024/189293789-1920c685-a1a0-42f5-99d6-01c146ba612f.png)

jerolimov replied:
My thought is that IST and ISI are resources, while External container image isn't. Wdyt?

invincibleJai commented:
/lgtm
